### PR TITLE
Book A Doctor Appointment app (Frontend)

### PR DESCRIPTION
**Pull request by katarighe**
Dear everyone, 😊🙏

We are excited to present our frontend pull request for the "Book A Doctor Appointment" app. 

The Book A Doctor appointments (Frontend) is a responsive React application that allows users to book appointments with doctors. This responsive application enables users to view, create, and delete both doctors and reservations. Access to the application requires users to sign up or sign in, which can be done through the provided forms.

Built with React, this project was a true collaborative joy. Here's a summary of our achievements:

- **React App Setup:** Established a new React app with proper linter configuration for code quality.
- **User Interface:** Designed and implemented user interfaces for both login and signup functionalities.
- **Authentication:** Implemented user authentication system to grant authorized access based on user roles.
- **Welcome Component:** Created a welcoming interface showcasing a catalog of doctors in a visually appealing slide format.
- **Doctor Management:** Developed interfaces for adding new doctors and managing their information.
- **Appointment Reservation:** Implemented a user-friendly interface for reserving appointments with preferred doctors.
- **Doctor List & Removal:** Built a dynamic list of doctors, including the ability to delete them.
- **Dynamic Navigation:** Implemented a role-based navigation bar that adapts based on the user's logged-in status.
- **Appointment Management:** Developed a dedicated view appointment component with a convenient cancellation function.

We deeply appreciate you taking the time to review our pull request.

For the backend pull request, please [click here](https://github.com/katarighe/book-appointment-back-end/pull/23).

Thank you for your careful consideration of our proposed changes. 🏅

### CODE REVIEW

File: src/components/Sidebar/Sidebar.scss [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        margin: 0;
+      }
+
+      @media screen and (max-width: 900px) {
+        .active,
+        .notActive {
+          padding: 15px;
+          display: flex;
+          // align-items: center;
+          // justify-content: center;
```
Comment (Meltrust) [inline src/components/Sidebar/Sidebar.scss]: - [ ] Kindly, remove the commented code present in this file in order to improve the repo's cleanliness. 👍

File: src/Hooks/fetchAllDataHook.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+// export const useFetchAllDataHooks = () => {
+//   const getJob = useCallback(() => {
+//     API.newJobs().then((res) => {
+//       console.log(res);
+//     });
+//   }, []);
+
+//   return { getJob };
+// };
```
Comment (Meltrust) [inline src/Hooks/fetchAllDataHook.js]: - [ ] Kindly, remove the commented code present in this file in order to improve the repo's cleanliness. 👍

File: src/App.jsx [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+import generalRoutes from './Layout/Routes/GeneralRutes';
+import dashboardRoutes from './Layout/Routes/DashboardRoutes';
+import DashboardLayout from './Layout/DashboardLayout';
+import NotFound from './Pages/404/NotFound';
+import { useEffect } from 'react';
+import { useAuthHook } from './Hooks/authHook';
+import { useSelector } from 'react-redux';
+import { selectUserData } from './Redux/Features/userAuthSlice';
+
+function App() {
```
Comment (Meltrust) [inline src/App.jsx]: - [ ]  Please, kindly refactor this main App functional component to ES6 arrow function, so your code complies with the latest standards.

Refactoring is easy:

```javascript
const App = () => {
  // ...
}
```

## Explanation

The `function` keyword has been soft deprecated in almost all the software industry in favor of ES6 arrow functions. This means, almost nobody uses it anymore, so much that it's considered a non-best practice.

- [ ] OPTIONAL: Refactor the rest of the functions.

You can do a search in Vscode for the "function" word, and correct as needed.  Shouldn't take you more than 5-10 min so don't worry:

image


That way, your code will meet the latest standards. 💪

File: src/Pages/BookAppointment/BookAppointments.jsx [hunk 4]
```diff
@@ -0,0 +1,79 @@
+
+import { Spinner } from 'react-bootstrap';
+import { useDispatch, useSelector } from 'react-redux';
+import { selectDoctor } from '../../Redux/Features/doctorSlice';
+import { Link } from 'react-router-dom';
+import './BookAppointment.scss';
+import { getAppointmentDetails } from '../../Redux/Features/appointmentSlice';
+
+
+function BookAppointments() {
```
Comment (Meltrust) [inline src/Pages/BookAppointment/BookAppointments.jsx]: - [ ] I was able to book in the past:

image

Kindly, add some validation so your users can only book for the future. 👍

**Review by Meltrust — CHANGES_REQUESTED** [review]
Hi,

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation, but you are almost there!






### To highlight:

-  Nice code organization ✔️ 
-  App is working well✔️ 
-  Good readme ✔️ 
-  Beautiful, gorgeous frontend ✔️ 



#### You are really close to finishing the Microverse program!! Keep it up! 👍👍👍

You can do it

### After implementing the requested changes, please submit another review request. ♻️
_Check the comments under the review._







Cheers and Happy coding!👏👏👏

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the previous reviews unless it is requested otherwise._**

------

File: src/App.jsx (hunk 3 above)
Comment (Errytagedesign) [reply src/App.jsx]: This is a very detailed review, Thank you

**Review by Meltrust — APPROVED** [review]
# **Wow, you did it 🎉**
Brilliant
## Thank you for the changes implemented 💪 🥇 ㊗️ 
#### _Unless you want to add more features, go ahead to your final presentation ⏩ ⏩ ⏩_ 
## You are about to finish the Microverse program.  You have come a long way!!!




### Good luck in the software industry!!  I'll see you there. ✨
### **Congratulations!!!!!!**  🎉
applause

### To highlight


- Superb job ✔️  



Cheers and Happy coding!👏👏👏


------

**Pull request merged**